src/main/java/com/google/sps/data/Event.java

```diff
@@ -0,0 +1,189 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.google.sps.data;
+ 
+/**
+ * Class that creates an event specific to a POI 
+ */
+public class Event {
+ 
+  // inputs
+  private String name;
+  private String address;
+  private String date;
+  // format: HHMM
+  private int startTime;
+  private int endTime;
+
+  //in mins
+  private int travelTime;
+
+  // fields to be saved for parsing
+  private String strStartTime;
+  private String strEndTime;
+ 
+  // class constants
+  private static final int HALFHOUR = 30;
+  private static final int HOUR = 60;
+  private static final int MINUTES_IN_A_DAY = 1440;
+  private static final int MIN_POSSIBLE_TIME = 0;
+
+  /**
+   * Constructor that takes in time spent at location
+   * 
+   * @param name name of the location (NOT address)
+   * @param address exact address of the POI
+   * @param date date from user input travel day. Cannot be null.
+   * @param startTime start of activity (HHMM format)
+   * @param travelTime time spent traveling to next location (minutes)
+   * @param timeAtLocation time spent at POI (minutes)
+   */
+  public Event(String name, String address, String date, int startTime, 
+              int travelTime, int timeAtLocation) {
+    this.name = name;
+    this.address = address;
+    this.date = date;
+    this.startTime = startTime;
+    this.travelTime = travelTime;
+    this.endTime = calculateEndTime(timeAtLocation);
+
+    createCalendarTimes();
+  }
+ 
+  /**
+   * Constructor that assume one hour default spent at location
+   * 
+   * @param name name of the location (NOT address)
+   * @param address exact address of the POI
+   * @param date date from user input travel day. Cannot be null.
+   * @param startTime start of activity (HHMM format)
+   * @param travelTime time spent traveling to next location (minutes)
+   */
+  public Event(String name, String address, String date, int startTime, int travelTime) {
+    this.name = name;
+    this.address = address;
+    this.date = date;
+    this.startTime = startTime;
+    this.travelTime = travelTime;
+    this.endTime = calculateEndTime(HOUR);
+
+    createCalendarTimes();
+  }
+
+  // function that sets start and end Time with correct string format
+  private void createCalendarTimes() {
+    this.strStartTime = createStrTime(this.date, this.startTime);
+    this.strEndTime = createStrTime(this.date, this.endTime);
+  }
+
+  /**
+   * function that builds string w correct format for calendar-script.js
+   * 
+   * @param date date from user input travel day. Cannot be null.
+   * @param time time to to start or end event. Must be within bounds
+   */
+  private static String createStrTime(String date, int time) {
+    if (date == null) {
+      throw new NullPointerException("Date cannot be null");
+    }
+
+    if (time < MIN_POSSIBLE_TIME) {
+      throw new IllegalArgumentException("Time cannot be less than 0");
```

Comment:
nit: use 'MIN_POSSIBLE_TIME' constant in the message here so that if MIN_POSSIBLE_TIME is ever changed this exception message still makes sense.